File: 3128. Right Triangles/RightTriangles.cs
public class Solution {
    public long NumberOfRightTriangles(int[][] grid) {
        int rowCount = grid.Length;
        int colCount = grid[0].Length;

        int[] rowSums = new int[rowCount];
        int[] colSums = new int[colCount];

        for (int i = 0; i < rowCount; i++) {
            for (int j = 0; j < colCount; j++) {
                if (grid[i][j] == 1) {
                    rowSums[i]++;
                    colSums[j]++;
                }
            }
        }

        long totalTriangles = 0;

        for (int i = 0; i < rowCount; i++) {
            for (int j = 0; j < colCount; j++) {
                if (grid[i][j] == 1) {

                    totalTriangles += (long)(rowSums[i] - 1) * (colSums[j] - 1);
                }
            }
        }

        return totalTriangles;
    }
}

File: 3128. Right Triangles/right_triangles.py
class Solution(object):
    def numberOfRightTriangles(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        rowCount = [0] * len(grid)
        colCount = [0] * len(grid[0])

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    rowCount[i] += 1
                    colCount[j] += 1

        numRightTriangles = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    numRightTriangles += (rowCount[i] - 1) * (colCount[j] - 1)

        return numRightTriangles